libs/src/services/contracts/ContractClient.js

```diff
@@ -36,7 +36,7 @@ class ContractClient {
   }
 
   /** Inits the contract if necessary */
-  async init () {
+  async init (selectNewEndpointOnRetry = true) {
```

Comment:
this is part 1 of bugfix - if we always select a new web3provider endpoint on retry, the code fails since CN is initted with only a single web3provider. also logically, this just makes more sense to not always force a provider re-selection on a contract client init op

will move to separate PR

Reply:
diff gone, on master now